build-release.sh

```diff
@@ -1,87 +1,102 @@
-
-	local BUILD_ENV=("${@:4}")
-	local BUILD_ARGS=(-o "dist/$DIR/k6${SUFFIX}" -trimpath)
-
-	if [ -n "$VERSION_DETAILS" ]; then
-		BUILD_ARGS+=(-ldflags "-X github.com/loadimpact/k6/lib/consts.VersionDetails=$VERSION_DETAILS")
-	fi
-
-	echo "- Building platform: ${ALIAS} (" "${BUILD_ENV[@]}" "go build" "${BUILD_ARGS[@]}" ")"
+    local BUILD_ENV=("${@:3}")
+    local BUILD_ARGS=(-o "${OUT_DIR}/${NAME}/k6${SUFFIX}" -trimpath)
 
-	# Clean out any old remnants of failed builds.
-	rm -rf "dist/$DIR"
-	mkdir -p "dist/$DIR"
+    if [ -n "$VERSION_DETAILS" ]; then
+        BUILD_ARGS+=(-ldflags "-X github.com/loadimpact/k6/lib/consts.VersionDetails=$VERSION_DETAILS")
+    fi
 
-	# Subshell to not mess with the current env vars or CWD
-	(
-		export "${BUILD_ENV[@]}"
+    echo "- Building platform: ${ALIAS} (" "${BUILD_ENV[@]}" "go build" "${BUILD_ARGS[@]}" ")"
 
-		# Build a binary
-	 	go build "${BUILD_ARGS[@]}"
+    mkdir -p "${OUT_DIR}/${NAME}"
 
-		# Archive it all, native format depends on the platform.
-		cd dist
-		make_archive "$FMT" "$DIR"
-	)
+    # Subshell to not mess with the current env vars or CWD
+    (
+        export "${BUILD_ENV[@]}"
+        # Build a binary
+         go build "${BUILD_ARGS[@]}"
+    )
+}
 
-	# Delete the source files.
-	rm -rf "dist/$DIR"
+package() {
+    local ALIAS="$1" FMT="$2"
+    local NAME="k6-${VERSION}-${ALIAS}"
+    echo "- Creating ${NAME}.${FMT} package..."
+    case $FMT in
+    deb|rpm)
+        # The go-bin-* tools expect the binary in /tmp/
+        [ ! -r /tmp/k6 ] && cp "${OUT_DIR}/${NAME}/k6" /tmp/k6
+        "go-bin-${FMT}" generate --file "packaging/${FMT}.json" -a amd64 \
+            --version "${VERSION#v}" -o "${OUT_DIR}/k6-${VERSION}-amd64.${FMT}"
+        ;;
+    tgz)
+        tar -C "${OUT_DIR}" -zcf "${OUT_DIR}/${NAME}.tar.gz" "$NAME"
+        ;;
+    zip)
+        (cd "${OUT_DIR}" && zip -rq9 - "$NAME") > "${OUT_DIR}/${NAME}.zip"
+        ;;
+    *)
+        echo "Unknown format: $FMT"
+        return 1
+        ;;
+    esac
 }
 
+CHECKSUM_FILE="k6-${VERSION}-checksums.txt"
 checksum() {
-	local CHECKSUM_FILE="k6-${VERSION}-checksums.txt"
-
-	if command -v sha256sum > /dev/null; then
-		CHECKSUM_CMD=("sha256sum")
-	elif command -v shasum > /dev/null; then
-		CHECKSUM_CMD=("shasum" "-a" "256")
-	else
-		echo "ERROR: unable to find a command to compute sha-256 hash"
-		return 1
-	fi
-
-	rm -f "dist/$CHECKSUM_FILE"
-	( cd dist && for x in *; do [ -f "$x" ] && "${CHECKSUM_CMD[@]}" -- "$x" >> "$CHECKSUM_FILE"; done )
+    if command -v sha256sum > /dev/null; then
+        CHECKSUM_CMD=("sha256sum")
+    elif command -v shasum > /dev/null; then
+        CHECKSUM_CMD=("shasum" "-a" "256")
+    else
+        echo "ERROR: unable to find a command to compute sha-256 hash"
+        exit 1
+    fi
+
+    echo "--- Generating checksum file..."
+    rm -f "${OUT_DIR}/$CHECKSUM_FILE"
+    (cd "$OUT_DIR" && find * -maxdepth 0 -type f | xargs "${CHECKSUM_CMD[@]}" > "$CHECKSUM_FILE")
```

Comment:
Running shellcheck on this file gives me 
```
    (cd "$OUT_DIR" && find * -maxdepth 0 -type f | xargs "${CHECKSUM_CMD[@]}" > "$CHECKSUM_FILE")
                      ^------------------------^ SC2038: Use -print0/-0 or -exec + to allow for non-alphanumeric filenames.
                           ^-- SC2035: Use ./*glob* or -- *glob* so names with dashes won't become options.

For more information:
  https://www.shellcheck.net/wiki/SC2038 -- Use -print0/-0 or -exec + to allo...
  https://www.shellcheck.net/wiki/SC2035 -- Use ./*glob* or -- *glob* so name...
```

Reply:
Thanks, I haven't adopted `shellcheck` yet, but it seems useful. Fixed in bd2b64da.